app/src/main/java/an/maguste/android/navier/movieslist/MovieAdapter.kt

```diff
@@ -43,19 +46,23 @@ class MovieViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
             .apply(RequestOptions().centerCrop())
             .into(binding.poster)
 
-        when {
-            movie.adult -> binding.ageRating.text = itemView.resources.getString(R.string.age_rating_default)
-            else -> binding.ageRating.visibility = View.INVISIBLE
+        when (movie.adult) {
+            true -> binding.ageRating.text =
+                itemView.resources.getString(R.string.age_rating_default)
+            false -> binding.ageRating.visibility = View.INVISIBLE
         }
```

Comment:
Нужно заменить на обычный if
```suggestion
        if (movie.adult) {
           itemView.resources.getString(R.string.age_rating_default)
        } else {
            binding.ageRating.visibility = View.INVISIBLE
        }
```

Reply:
При компиляции в java код добавляется лишняя проверка
```
      boolean var2 = movie.getAdult();
      TextView var8;
      View var10001;
      if (var2) {
         var8 = this.binding.ageRating;
         Intrinsics.checkNotNullExpressionValue(var8, "binding.ageRating");
         var10001 = this.itemView;
         Intrinsics.checkNotNullExpressionValue(var10001, "itemView");
         var8.setText((CharSequence)var10001.getResources().getString(1900094));
      } else if (!var2) {
         var8 = this.binding.ageRating;
         Intrinsics.checkNotNullExpressionValue(var8, "binding.ageRating");
         var8.setVisibility(4);
      }
```